File: samples/snippets/fsharp/VS_Snippets_CLR_System/System.ArgumentOutOfRangeException/fs/NoElements.fs
module NoElements

// <Snippet4>
open System


let list = ResizeArray<string>()
printfn $"Number of items: {list.Count}"
try
    printfn $"The first item: '{list[0]}'"
with 
| :? ArgumentOutOfRangeException as e ->
    printfn $"{e.Message}"

// The example displays the following output:
//   Number of items: 0
//   Index was out of range. Must be non-negative and less than the size of the collection. (Parameter 'index')
// </Snippet4>

module Example2 = 
    let test () =
        let list = ResizeArray<string>()
        printfn $"Number of items: {list.Count}"
        // <Snippet5>
        if list.Count > 0 then
            printfn $"The first item: '{list[0]}'"
        // </Snippet5>

File: snippets/csharp/System/ArgumentOutOfRangeException/Overview/NoElements.cs
﻿// <Snippet4>
using System;
using System.Collections.Generic;

public class Example4
{
   public static void Main()
   {
      var list = new List<string>();
      Console.WriteLine("Number of items: {0}", list.Count);
      try {
         Console.WriteLine("The first item: '{0}'", list[0]);
      }
      catch (ArgumentOutOfRangeException e) {
         Console.WriteLine(e.Message);
      }
   }
}
// The example displays the following output:
//   Number of items: 0
//   Index was out of range. Must be non-negative and less than the size of the collection.
//   Parameter name: index
// </Snippet4>

public class Example5
{
   public static void Test()
   {
      var list = new List<string>();
      Console.WriteLine("Number of items: {0}", list.Count);
      // <Snippet5>
      if (list.Count > 0)
         Console.WriteLine("The first item: '{0}'", list[0]);
      // </Snippet5>
   }
}
